src/palace/manager/celery/tasks/playtime_entries.py

```diff
@@ -238,22 +237,24 @@ def generate_playtime_report(
                 )
 
 
-def _fetch_distinct_data_source_names_in_range(
-    session: Session, start: date, until: date
-) -> Query[str]:
-    return session.query(
-        select(
-            distinct(PlaytimeSummary.data_source_name),
+def _fetch_distinct_eligible_data_source_names(
+    session: Session,
+) -> list[str]:
+    query = (
+        select(Collection)
+        .join(
+            IntegrationConfiguration,
+            Collection.integration_configuration_id == IntegrationConfiguration.id,
         )
         .where(
-            and_(
-                PlaytimeSummary.timestamp >= start,
-                PlaytimeSummary.timestamp < until,
+            IntegrationConfiguration.protocol.in_(
+                [OPDS2WithODLApi.label(), OPDSForDistributorsAPI.label()]
```

Comment:
Change requested: I believe we'll need to use some combination of the previous query and this new one to come up with a complete list of datasources for all expected scenarios. This is because a collection might be delete after some reading, but before the report is run. That's why we capture string versions -- and not just table IDs -- for some of the properties. We don't want to lose accounting if certain objects are removed from the database.